ext/pdo/pdo_stmt.c

```diff
@@ -1031,7 +1031,7 @@ static int do_fetch(pdo_stmt_t *stmt, int do_bind, zval *return_value, enum pdo_
 					if (Z_REFCOUNTED(val)) {
 						Z_ADDREF(val);
 					}
-					zend_hash_next_index_insert(Z_ARRVAL_P(return_value), &val);
+					zend_hash_index_add(Z_ARRVAL_P(return_value), i, &val);
```

Comment:
This call may now fail (if a column name equals a column index), in which case the above refcount increment may leak.

Reply:
Thanks! Fixed with 7209b4fc3684588d0af916a598358cdc9af37bba.